ethereum/sidecar/server.go

```diff
@@ -953,11 +981,106 @@ func (s *Server) fetchAssetsUnlockConfirmedEvents(
 	return abiEvents, nil
 }
 
-//nolint:unparam
-func (s *Server) attestAssetsUnlockedEvents(ctx context.Context) error {
-	// TODO: Implement
-	<-ctx.Done()
-	return nil
+func (s *Server) unqueueAttestation() *bridgetypes.AssetsUnlockedEvent {
+	s.attestationMutex.Lock()
+	defer s.attestationMutex.Unlock()
+	if len(s.attestationQueue) == 0 {
+		return nil
+	}
+
+	attestation := s.attestationQueue[0]
+	s.attestationQueue = s.attestationQueue[1:]
+
+	return &attestation
+}
+
+func (s *Server) attestAssetsUnlockedEvents(ctx context.Context) {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// get attestations if any
+			for attestation := s.unqueueAttestation(); attestation != nil; attestation = s.unqueueAttestation() {
+
+				// try to execute this transaction until we have done so successfully.
+				retry := true
+				for retry && s.shouldAttest(attestation) {
+					// TODO: first verify that:
+					// - we haven't attested this yet
+					// - it's not been already confirmed via enough validators attestation yet
+
+					if err := s.txExecutor.Send(attestation); err != nil {
+						s.logger.Error("error sending attestation %cv to MezoBridge: %v, rescheduling to be executed later", attestation.String(), err)
+						continue
+					}
+
+					retry = false
+				}
+
+			}
+		case <-ctx.Done():
+			s.logger.Info(
+				"stopping assets unlocked attestations to context cancellation",
+			)
+		}
+	}
+}
+
+func (s *Server) shouldAttest(attestation *bridgetypes.AssetsUnlockedEvent) bool {
+	callOpts := &bind.CallOpts{
+		BlockNumber: big.NewInt(-4), // this is safe block
```

Comment:
I believe -4 is a nice constant in go-ethereum, let's use it. BTW, shouldn't we use **finalized** block?